### PR TITLE
Fix compiler warning

### PR DESCRIPTION
**Pull request by NiklasWik**
Related to https://github.com/dealii/dealii/issues/13910#issuecomment-1147340404

Haven't tried to compile myself but I'm guessing this would work.

### CODE REVIEW

File: include/deal.II/matrix_free/evaluation_kernels.h [hunk 1]
```diff
@@ -3697,6 +3697,9 @@ namespace internal
       const unsigned int                            face_no,
       const MatrixFreeFunctions::ShapeInfo<Number> &shape_info)
     {
+      if (dim == 1)
```
Comment (peterrum) [inline include/deal.II/matrix_free/evaluation_kernels.h:3700]: Question: is it possible to have a 1D RT? If not one should probably also add here an assert!

Comment (NiklasWik) [reply include/deal.II/matrix_free/evaluation_kernels.h:3700]: I think it is asserted in the class itself, but I might be mistaken

Comment (peterrum) [reply include/deal.II/matrix_free/evaluation_kernels.h:3700]: There is no assert here:

https://github.com/dealii/dealii/blob/ecdd5f268b0da3a1ae4c23d974fab1b870f249f1/include/deal.II/matrix_free/evaluation_kernels.h#L3555-L3557

Comment (NiklasWik) [reply include/deal.II/matrix_free/evaluation_kernels.h:3700]: This is the one I was referring to 
https://github.com/dealii/dealii/blob/ecdd5f268b0da3a1ae4c23d974fab1b870f249f1/source/fe/fe_raviart_thomas_nodal.cc#L445

**Review by kronbichler — APPROVED** [review]
This looks good, let us merge it soon and see if the outcome is as desired.

File: include/deal.II/matrix_free/evaluation_kernels.h [hunk 2]
```diff
@@ -3697,6 +3697,9 @@ namespace internal
...
+        return;
```
Comment (marcfehling) [inline include/deal.II/matrix_free/evaluation_kernels.h:3701]:
```suggestion
      if (dim == 1)
        {
          Assert(false, ExcInternalError());
          return;
        }
```
As you said, as this condition should never be met, I would suggest to clarify this be adding another assertion here right before the return statement. If you want to, you could add another in-source comment to why this return statement is necessary.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**